mmdet/evaluation/metrics/crowdhuman_metric.py

```diff
@@ -0,0 +1,820 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+import json
+import os.path as osp
+import tempfile
+from collections import OrderedDict
+from multiprocessing import Process, Queue
+from typing import Dict, List, Optional, Sequence, Union
+
+import numpy as np
+
+try:
+    from scipy.sparse import csr_matrix
+    from scipy.sparse.csgraph import maximum_bipartite_matching
+except ImportError:
+    csr_matrix, maximum_bipartite_matching = None, None
+
+from mmengine.evaluator import BaseMetric
+from mmengine.fileio import FileClient, dump, load
+from mmengine.logging import MMLogger
+
+from mmdet.evaluation.functional.bbox_overlaps import bbox_overlaps
+from mmdet.registry import METRICS
+
+PERSON_CLASSES = ['background', 'person']
+
+
+@METRICS.register_module()
+class CrowdHumanMetric(BaseMetric):
+    """CrowdHuman evaluation metric.
+
+    Evaluate AP, MR and JI for detection tasks.
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        metric (str | List[str]): Metrics to be evaluated. Valid metrics
+            include 'AP', 'MR' and 'JI'. Defaults to 'AP'.
+        format_only (bool): Format the output results without perform
+            evaluation. It is useful when you want to format the result
+            to a specific format and submit it to the test server.
+            Defaults to False.
+        outfile_prefix (str, optional): The prefix of json files. It includes
+            the file path and the prefix of filename, e.g., "a/b/prefix".
+            If not specified, a temp file will be created. Defaults to None.
+        file_client_args (dict): Arguments to instantiate a FileClient.
+            See :class:`mmengine.fileio.FileClient` for details.
+            Defaults to ``dict(backend='disk')``.
+        collect_device (str): Device name used for collecting results from
+            different ranks during distributed training. Must be 'cpu' or
+            'gpu'. Defaults to 'cpu'.
+        prefix (str, optional): The prefix that will be added in the metric
+            names to disambiguate homonymous metrics of different evaluators.
+            If prefix is not provided in the argument, self.default_prefix
+            will be used instead. Defaults to None.
+        eval_mode (int): Select the mode of evaluate. Valid mode include
+            0(just body box), 1(just head box) and 2(both of them).
+            Defaults to 0.
+        iou_thres (float): IoU threshold. Defaults to 0.5.
+        compare_matching_method (str, optional): Matching method to compare
+            the detection results with the ground_truth when compute 'AP'
+            and 'MR'.Valid method include VOC and None(CALTECH). Default to
+            None.
+        mr_ref (str): Different parameter selection to calculate MR. Valid
+            ref include CALTECH_-2 and CALTECH_-4. Defaults to CALTECH_-2.
+        ji_process_num (int): The number of processes to evaluation JI.
```

Comment:
Should explain what is JI and MR in the docstring. We can say:
`ji_process_num (int): The number of processes to evaluate Jaccard Index (JI).`
Also suggest using num_ji_process to follow the convention of `num_xxx`.